### DISCUSSION TITLE
Eglot issues with python-lsp-server:

### DISCUSSION THREAD

**Discussion by zeta-00** [joaotavora/eglot#913] [Q&A]
Greetings,

I'm having an issue where Eglot is not working with python-lsp-server, sometimes Eglot dies and throws the missing pkgs_resources error(I have setup tools install in macos, so I'm not sure what's wrong), and when I try:

`(add-to-list 'eglot-server-programs '(python-mode . ("python-lsp-server")))`, 

Eglot is not able to find `python-lsp-server` binary PATH?

If you could give me some tips on how to fix this issue.

https://github.com/python-lsp/python-lsp-server/issues/157

python3 -m pylsp --check-parent-process

Thanks in advance for the help.

**Comment by zeta-00** [comment]
Note I'm using an M1 mac(aarch64-darwin), and with Emacs I'm using `use-package`.

**Comment by zeta-00** [comment]
;;---------------------------------------------------
(use-package eglot
  :defer t
  :hook
  (python-mode . eglot-ensure))
;;---------------------------------------------------
;; (add-to-list 'eglot-server-programs '(python-mode . ("python-lsp-server")))
;;---------------------------------------------------

**Comment by zeta-00** [comment]
Also, I'm using the Nix package manager to handle the Python packages. The hash id of the Python packages changes because of this, so I'm wondering if there's a more generic way to declare the `python-lsp-server` binary PATH?

**Comment by zeta-00** [comment]
This link has some good documentation, but, it's specific to `lsp-mode` and Microsoft's `python-language-server`, so I'm not sure how to do something similar with `Eglot and Nix`.

**Comment by TLATER** [comment]
As someone using `eglot` + `python-lsp-server` with nix and nix shells, I don't need to set `eglot-server-programs` at all - it just resolves the binary from `$PATH` by default.

Note that the binary for that project isn't called `python-lsp-server`, but `pylsp`. If you don't set `eglot-server-programs`, it should resolve that by default, assuming it actually is on your `$PATH`; your configuration overrides the default behavior and presumably breaks it because the binary name is now wrong.
